### PR TITLE
OCR2VRF DKG Plugin

### PR DESCRIPTION
**Pull request by makramkd**
This PR:

* Adds the OCR2VRF DKG plugin as an OCR2 job type,
* Adds scripts to deploy the associated contracts,
* Adds automation scripts to set up a local DKG OCR2 cluster

### CODE REVIEW

File: core/cmd/dkg_configure_command.go [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+type DKGTemplateArgs struct {
+	contractID              string
+	ocrKeyBundleID          string
+	p2pv2BootstrapperPeerID string
+	p2pv2BootstrapperPort   string
+	transmitterID           string
+	chainID                 int64
+	EncryptionPublicKey     string
+	KeyID                   string
+	SigningPublicKey        string
```
Comment (makramkd) [inline core/cmd/dkg_configure_command.go:40]: Can make these fields private.

File: core/cmd/dkg_configure_command.go [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+	OnChainPublicKey  string
+	OffChainPublicKey string
+	ConfigPublicKey   string
+	PeerID            string
+	Transmitter       string
+	DkgEncrypt        string
+	DkgSign           string
+}
+
+type DKGTemplateArgs struct {
```
Comment (makramkd) [inline core/cmd/dkg_configure_command.go:31]: Can make this struct private.

File: core/scripts/ocr2vrf/setup_dkg.go [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+	"github.com/smartcontractkit/chainlink/core/logger"
+	helpers "github.com/smartcontractkit/chainlink/core/scripts/common"
+	"github.com/urfave/cli"
+)
+
+func setupDKGNodes(e helpers.Environment) {
+	client := NewProductionClient()
+	app := cmd.NewApp(client)
+
+	// Parse os arguments.
```
Comment (makramkd) [inline core/scripts/ocr2vrf/setup_dkg.go:24]: Can remove this comment.

File: core/cmd/dkg_configure_command.go [hunk 4]
```diff
@@ -0,0 +1,271 @@
...
+	if err != nil {
+		return errors.Wrap(err, "failed to unmarshal job spec")
+	}
+	jb.BootstrapSpec = &os
+
+	err = app.AddJobV2(context.Background(), &jb)
+	if err != nil {
+		return errors.Wrap(err, "failed to add job")
+	}
+	fmt.Println(sp)
```
Comment (jinhoonbang) [inline core/cmd/dkg_configure_command.go:230]: nit-pick: use logger instead

File: core/cmd/dkg_configure_command.go [hunk 5]
```diff
@@ -0,0 +1,271 @@
...
+	if err != nil {
+		return errors.Wrap(err, "failed to unmarshal job spec")
+	}
+	jb.OCR2OracleSpec = &os
+
+	err = app.AddJobV2(context.Background(), &jb)
+	if err != nil {
+		return errors.Wrap(err, "failed to add job")
+	}
+	fmt.Println(sp)
```
Comment (jinhoonbang) [inline core/cmd/dkg_configure_command.go:268]: same as above

**Review by jinhoonbang — APPROVED** [review]

**Review by jinhoonbang — APPROVED** [review]

**Pull request merged**